### PR TITLE
Add code samples for creational patterns

### PR DESCRIPTION
**Pull request by alexncr2**

### CODE REVIEW

File: OOP Fundamentals and C#/09 - Design Patterns - Creational/Singleton/SafePrinter.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        private SafePrinter()
+        {
+        }
+
+        public static SafePrinter Instance
+        {
+            get
+            {
+                lock (padlock)
```
Comment (AdinaMoldovan) [inline OOP Fundamentals and C#/09 - Design Patterns - Creational/Singleton/SafePrinter.cs]: I would suggest here to add one more if(instance == null)) before executing the lock. Lock is an expensive operation. and there's no need for it to be performed every time we need an instance. Maybe it is not really relevant for the topic of the lesson, but if we show this, I would say it's better to show it correctly

Comment (alexncr2) [reply OOP Fundamentals and C#/09 - Design Patterns - Creational/Singleton/SafePrinter.cs]: added one more null check

**Review by AdinaMoldovan — APPROVED** [review]

File: OOP Fundamentals and C#/09 - Design Patterns - Creational/Singleton/Program.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+﻿using Singleton;
+
```
Comment (OlegLukash) [inline OOP Fundamentals and C#/09 - Design Patterns - Creational/Singleton/Program.cs:2]: Could we also add a static constructor version of singleton, please

Comment (alexncr2) [reply OOP Fundamentals and C#/09 - Design Patterns - Creational/Singleton/Program.cs:2]: added example using static

**Review by OlegLukash — APPROVED** [review]

**Pull request merged**